### PR TITLE
[Translation] Use proven DSN class from Notifier and rename package

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | ---

This class is already in use, no need to introduce it as experimental

cc @welcoMattic

### CODE REVIEW

**Comment by Nyholm** [comment]
Could you have a look at the psalm issues too?

**Comment by welcoMattic** [comment]
We already mention it, but we have many DSN implementation in the core. @Nyholm wrote a Dsn Component. I could be better for us to work together for a single Dsn implementation for 6.0? cc @nicolas-grekas

**Comment by OskarStark** [comment]
Sure we can do this but the DSN on notifier fixed some issues which mailer DSN still has because of BC, so we should copy this one.

**Review by welcoMattic — APPROVED** [review]

File: src/Symfony/Component/Translation/Bridge/Loco/composer.json
Comment (OskarStark) [inline src/Symfony/Component/Translation/Bridge/Loco/composer.json:2]: @fabpot we suffix the notifier with `notifier`, `-translation-provider` is more clear than `-translation`

File: .github/composer-config.json
Comment (OskarStark) [inline .github/composer-config.json:9]: I added this, because my tests pass on local:
CleanShot 2021-04-26 at 09 07 11

File: src/Symfony/Component/Translation/Bridge/Loco/composer.json
Comment (OskarStark) [reply src/Symfony/Component/Translation/Bridge/Loco/composer.json:2]: cc @welcoMattic

Comment (fabpot) [reply src/Symfony/Component/Translation/Bridge/Loco/composer.json:2]: `loco-translator`?

Comment (fabpot) [reply src/Symfony/Component/Translation/Bridge/Loco/composer.json:2]: I agree that adding `-translation-provider` to the provider name looks better.

**Comment by OskarStark** [comment]
Ready to merge from my side

File: .github/composer-config.json
Comment (stof) [reply .github/composer-config.json:9]: The right fix is to avoid relying on files which are inside the `Tests` folder of symfony/translation if they are meant to be exposed to other packages. Shared test cases should be in the `Test` namespace instead.

Comment (OskarStark) [reply .github/composer-config.json:9]: Makes sense will move them to Test

**Review by welcoMattic — APPROVED** [review]
Thank you for these improvements

File: .github/composer-config.json
Comment (OskarStark) [reply .github/composer-config.json:9]: Hmm I removed it, but I don't rely on sth in `Tests`, and the build was red, I readded this, maybe it is because even `Translation/Test/....TestCase` is not released yet and we depend on 5.3 🤔  That at least was the case when I was dealing with Notifier bridges

**Comment by OskarStark** [comment]
Please see my comment: https://github.com/symfony/symfony/pull/40944/files#r621020058

File: .github/composer-config.json
Comment (stof) [reply .github/composer-config.json:9]: do you have a link to a failing build ? Maybe that was a case where the component-level composer installation was using a symfony/translation package from packagist rather than the patched version from your PR.

**Review by Nyholm — APPROVED** [review]
Thank you

**Comment by Nyholm** [comment]
Oh, a self-merge =)

**Comment by stof** [comment]
@Nyholm self-merge are OKs as long as you have the right number of approvals from the core team.

**Comment by Nyholm** [comment]
Yeah, I know they are OK. But isn't it at least some stigma about it?

**Pull request merged**
